### DISCUSSION TITLE
SLSPC questions

### DISCUSSION THREAD

**Discussion by martinholmer** [PolicyEngine/policyengine-us#2024] [Q&A]
I'm trying to write some integration tests involving the federal Premium Tax Credit.
However, I'm not making much progress because I'm puzzled about a few things.
Here are three questions that reveal my lack of knowledge in this area:

*  **Why are NY and VT missing from the following SLSPC parameter file?**
`data/parameters/gov/hhs/medicaid/geography/second_lowest_silver_plan_cost.yaml`

* **Why do passing tests in source code tree, fail when using just a package?**
Aren't the SLSPC data incorporated in packages made from the source code?
If those data are incorporated in the package, is the model flexible enough
to import SLSPC data from either the source code or a package?  See the 
error message below for details.

* **Why is a state_code needed when a zip_code is supplied?**
Every zip code falls into a single state, right?
When specifying a Buffalo, NY, zip code (14203) with no state code, why does
the model think the household lives in California?

Here is the error message I get when I run an integration test (that passes when
it is located in the `policyengine_us/tests/policy/baseline` source code tree) 
outside of the source code tree (in the presence of an otherwise fully operational
`policyengine_us` package):
```
E   Hint: make sure your test modules/packages have valid Python names.
E   Traceback:
E   __init__.py:7: in <module>
E       from .data import *
E   E   ImportError: attempted relative import with no known parent package
=========================== short test summary info ============================
ERROR ptc.yaml:: - _pytest.nodes.Collector.CollectError: ImportError while im...
=============================== 1 error in 0.62s ===============================
```
It looks as if the `policyengine_us/data/parameters/...` tree is not in the package.
Why isn't the `second_lowest_silver_plan_cost.yaml` file in the regular
`policyengine_us/parameter/gov/...` tree?

**Comment by MaxGhenis** [comment]
@nikhilwoodruff is the expert here but just on the third point, I don't think any reason in particular except we haven't brought in a ZIP-state mapping

**Reply by martinholmer** [reply]
Are there any plans to add a ZIP-to-state mapping?  If not, there should be somewhere a warning to users that a zip code is not sufficient to determine the households state.

**Comment by MaxGhenis** [comment]
Also the location of the file is to avoid loading it (the biggest parameter we have by far) into the web app, which slows it down considerably

**Reply by martinholmer** [reply]
Where are the SLSPC data then?  Never included in a policyengine-us package?  Or are they somehow included in the CPS data files (as suggested by recent revisions to the CPS file creation code)?   If they are in the CPS data, why can't a local package find them there?  It was my impression that when you make a package, the CPS data file is included.  Is that not correct?
